code-dot-org/code-dot-org
# [Google Blockly][Utils] Move setHSV to utils

Right now, we have Blockly classes that contain functions that can be pulled out as utility functions. In order to better understand exactly what google Blockly functionality we’re overriding in our wrapper classes, we need to move all utility-type functions out of the classes. 

 `setHSV` is a utility function used to set the color of a Blockly block. 

Acceptance Criteria: 

- Move `setHSV` to a utility function 
- Update how we are calling the function in all blockly labs (google blockly and cdo blockly) 

QA Notes: 

- Check for regressions in live google blockly V6 labs.
- Check for regressions in cdo blockly labs

## Links


- jira ticket: [STAR-2111: [Google Blockly][Utils] Move setHSV to utils](https://codedotorg.atlassian.net/browse/STAR-2111)


## Testing story

Throughout testing, all relevant labs were loaded to confirm nothing was breaking. Temporarily modifying raw HSV values for specific blocks resulted in expected changes, signaling that the the code changes here are the right ones.

Tested after each commit in a variety of Blockly labs.
*Links are for a local server, for the convenience of any reviewer who wants to pull these changes.*
* [Poetry](http://localhost-studio.code.org:3000/s/poetry-2021/lessons/7/levels/1)
* [Flappy](http://localhost-studio.code.org:3000/flappy/10)
* [Sprite Lab](http://localhost-studio.code.org:3000/projects/spritelab)
* [Dance](http://localhost-studio.code.org:3000/s/dance-2019/lessons/1/levels/10)
* [Bounce](http://localhost-studio.code.org:3000/s/sports/lessons/1/levels/8)
* Minecraft:
  * [Simple](http://localhost-studio.code.org:3000/s/mc/lessons/1/levels/14)
  * [Designer](http://localhost-studio.code.org:3000/s/minecraft/lessons/1/levels/12)
  * [Agent](http://localhost-studio.code.org:3000/s/hero/lessons/1/levels/12)
  * [Code Connection](http://localhost-studio.code.org:3000/projects/minecraft_codebuilder)
* Maze:
  * [Bee](http://localhost-studio.code.org:3000/s/express-2021/lessons/15/levels/13)
  * [Collector](http://localhost-studio.code.org:3000/levels?utf8=%E2%9C%93&name=grade3_RunningFarm_8) - *not used in any curricula*
  * [Harvester](http://localhost-studio.code.org:3000/s/express-2021/lessons/19/levels/10)
  * [Planter](http://localhost-studio.code.org:3000/levels?utf8=%E2%9C%93&name=Planter) - *not used in any curricula*
* [Play Lab](http://localhost-studio.code.org:3000/s/playlab/lessons/1/levels/10)
* [Artist](http://localhost-studio.code.org:3000/s/express-2021/lessons/21/levels/13)
* [Jigsaw](http://localhost-studio.code.org:3000/s/jigsaw/lessons/1/levels/10)
* [Eval/Calc](http://localhost-studio.code.org:3000/s/algebra/lessons/14/levels/10)

## Follow-up work

Continue moving all utility-type functions out of the Blockly classes.
